src/test/socket/accept/test_accept.rs

```diff
@@ -24,32 +24,53 @@ fn main() -> Result<(), String> {
     // should we summarize the results rather than exit on a failed test
     let summarize = std::env::args().any(|x| x == "--summarize");
 
-    let tests = if run_only_passing_tests {
-        get_passing_tests()
-    } else {
-        get_all_tests()
-    };
+    let mut tests = get_tests();
+    if run_only_passing_tests {
+        tests = tests
+            .into_iter()
+            .filter(|x| x.shadow_passing() == test_utils::ShadowPassing::Yes)
+            .collect()
+    }
 
-    test_utils::run_tests(&tests, summarize)?;
```

Comment:
Why pull this conditional out of test_utils::run_tests?

Reply:
I did that since when `summarize == true`, it will never return an `Err` even if a test failed. I didn't want the caller to think that the tests passed just because it returned `Ok`. I have changed it back to how it was before though since it was overly complex.